app/models/reservation.rb

```diff
@@ -120,6 +120,9 @@ def self.upcoming_offline(start_at_limit)
       .where(product_id: OfflineReservation.current.pluck(:product_id))
       .not_canceled
       .not_ended
+      .joins(:order_detail)
+      .where("order_details.state" => %w(new inprocess))
+      .where("order_details.problem" => false)
```

Comment:
I forget exactly how it works, but I think the all the jobs move the order to completed. I would double check that.

I _think_ you can do `where(order_details: { state: %w(new inprocess), problem: false })`. Also, does this scope ignore un-purchased reservations sitting in carts? We could probably use `merge(OrderDetail.new_or_inprocess)` which includes that check as well.